src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -220,9 +220,9 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 
 	private locked = false;
 
-	private selection: EditorInput[] = [];					// editors in selected state, first one is active
+	private selection: EditorInput[] = [];					// editors in selected state, first one is active TODO align this with transient editors and use a Set<EditorInput> instead
```

Comment:
@benibenj I would align this with `transient` editors and use a `Set<EditorInput>`

Reply:
But then I don't know which editor is the active editor. This would mean there would need to be a property which stores that leading to active editor and selection potentially not being in sync

Reply:
Hm yeah, its not impossible. `Set` is ordered and so the first element would always stay the same and can be accessed, but only via some call such as `values().next()`, which should be relatively cheap since its an `Iterator`, but yeah probably slower than array access.

Reply:
![image](https://github.com/microsoft/vscode/assets/900690/22833140-7e2c-4950-8406-e0fc5b5ed7e5)

Reply:
So let's keep array. The `setSelection` API makes sure to dedupe the selection.